src/api/helpers/get-expiring-applications/date-checks.test.ts

```diff
@@ -0,0 +1,156 @@
+
+      expect(result.length).toEqual(1);
+    });
+  });
+
+  describe('31st December', () => {
+    it('should return an array with 1 application', async () => {
+      const date = '2024-12-31';
+
+      jest.setSystemTime(new Date(date));
+
+      const updateData = generateUpdateData(date);
+
+      await applications.update({ context, applicationId: application.id, data: updateData });
+      application = await applications.get({ context, applicationId: application.id });
+
+      const result = await getExpiringApplications(context);
+
+      expect(result.length).toEqual(1);
+    });
+  });
+
+  describe('Deadline on 1st of month', () => {
+    it('should return an array with 1 application', async () => {
+      const date = '2024-03-30';
+
+      jest.setSystemTime(new Date(date));
+
+      const updateData = generateUpdateData(date);
+
+      await applications.update({ context, applicationId: application.id, data: updateData });
+      application = await applications.get({ context, applicationId: application.id });
+
+      const result = await getExpiringApplications(context);
+
+      expect(result.length).toEqual(1);
+    });
+  });
+
+  describe('Deadline on 2nd of month', () => {
+    it('should return an array with 1 application', async () => {
+      const date = '2024-03-31';
+
+      jest.setSystemTime(new Date(date));
+
+      const updateData = generateUpdateData(date);
+
+      await applications.update({ context, applicationId: application.id, data: updateData });
+      application = await applications.get({ context, applicationId: application.id });
+
+      const result = await getExpiringApplications(context);
+
+      expect(result.length).toEqual(1);
+    });
+  });
+
+  describe('Deadline 1 day ahead and end of the month', () => {
+    it('should return an empty array', async () => {
+      const date = '2024-04-01';
+
+      jest.setSystemTime(new Date(date));
+
+      const updateDate = '2024-03-31';
+      const updateData = generateUpdateData(updateDate);
+
+      await applications.update({ context, applicationId: application.id, data: updateData });
+      application = await applications.get({ context, applicationId: application.id });
+
+      const result = await getExpiringApplications(context);
+
+      expect(result.length).toEqual(0);
+    });
+  });
+
+  describe('Deadline 3 days ahead and end of the month', () => {
+    it('should return an empty array', async () => {
+      const date = '2024-04-02';
+
+      jest.setSystemTime(new Date(date));
+
+      const updateDate = '2024-03-30';
+      const updateData = generateUpdateData(updateDate);
+
+      await applications.update({ context, applicationId: application.id, data: updateData });
+      application = await applications.get({ context, applicationId: application.id });
+
+      const result = await getExpiringApplications(context);
+
+      expect(result.length).toEqual(0);
+    });
+  });
```

Comment:
Nice use of `jest.setSystemTime`

Could we add a test for deadline being the same day as now?